src/conn/conn_tiered.c

```diff
@@ -194,8 +197,21 @@ __wt_tier_do_flush(
     WT_RET(storage_source->ss_flush(
       storage_source, &session->iface, bucket_fs, local_name, obj_name, NULL));
 
-    WT_WITH_CHECKPOINT_LOCK(session,
-      WT_WITH_SCHEMA_LOCK(session, ret = __tier_flush_meta(session, tiered, local_uri, obj_uri)));
+    /*
+     * Flushing the metadata grabs the data handle with exclusive access, and the data handle may be
+     * held by the thread that queues the flush tier work item. As a result, the handle may be busy,
+     * so retry as needed, up to a few seconds.
+     */
+    for (tries = 1, millis = 100; tries <= 6; tries++) {
```

Comment:
There is a `WT_RETRY_MAX` define used in a few places that is 10. I suggest using that.

Reply:
A potential problem with letting the number of (re)tries go to 10 is that the last sleep would be 100 seconds, which seems rather long for this type of race.  Obviously you could start with a shorter sleep, or keep the current retry count.  Either way it seems like guesswork until we start using the system for real and see how often this race happens and how long it lasts.

Reply:
I would probably adjust the multiplier first. Try the 10 times at a fixed, short interval.

Reply:
Since msecs is doubling each time, I'm a little wary of bumping to 10, that makes the last wait 50 seconds.  I changed the code to exit the loop when the msecs > a few seconds, matching the comment.